### PR TITLE
Replace the Release bot PAT with GitHub App

### PR DESCRIPTION
**Pull request by Jeffreyhung**
Using this as a proof of concenpt for https://www.notion.so/sentry/DACI-Replace-GitHub-bot-accounts-with-GitHub-Apps-getsentry-release-15109965d1204a91b9be71c49e8b66e0

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -15,14 +15,25 @@ jobs:
...
-          GITHUB_TOKEN: ${{ secrets.GH_RELEASE_PAT }}
+          GITHUB_TOKEN: ${{ steps.token.outputs.token }}
         with:
           version: ${{ github.event.inputs.version }}
           force: ${{ github.event.inputs.force }}
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (Jeffreyhung) [reply .github/workflows/release.yml]: That is for sure by accident 😅 removed!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -15,14 +15,20 @@ jobs:
     runs-on: ubuntu-latest
     name: "Release a new version"
     steps:
+      - name: Get auth token
+        id: token
+        uses: actions/create-github-app-token@3378cda945da322a8db4b193e19d46352ebe2de5 #v.1.10.4
```
Comment (asottile-sentry) [inline .github/workflows/release.yml:20]: hmm the tag here doesn't look right

**Review by asottile-sentry — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (Jeffreyhung) [reply .github/workflows/release.yml:20]: I'm using the one from maintained by GitHub https://github.com/actions/create-github-app-token

Comment (asottile-sentry) [reply .github/workflows/release.yml:20]: yes but `v.1.10.4` looks like a mistake

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -15,14 +15,20 @@ jobs:
     runs-on: ubuntu-latest
     name: "Release a new version"
     steps:
+      - name: Get auth token
+        id: token
+        uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 #v.1.11.0
```
Comment (asottile-sentry) [inline .github/workflows/release.yml:20]:
```suggestion
        uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 #v1.11.0
```

ok maybe it doesn't look weird to you but this is what I mean

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**